### PR TITLE
Add support for evidence with same name and host

### PR DESCRIPTION
**Pull request by aapomm**
### Spec
Nexpose can have multiple <node> entries in the output referring to the same host/ip. With the current implementation, if the similar <node> entries point to the same vulnerability, only 1 evidence will be created. This is caused by this line: https://github.com/dradis/dradis-nexpose/blob/main/lib/dradis/plugins/nexpose/formats/full.rb#L109
```
evidence[test_id][nexpose_node.address] = service_test
```
and evidence[][] can only hold one of the same vuln_id and host ip.

**Proposed solution**
Update the evidence to be an array to support multiple evidence of the same vuln/host.

### Other Information

If there's anything else that's important and relevant to your pull
request, mention that information here. This could include
benchmarks, or other information.

Thanks for contributing to Dradis!

### Copyright assignment

Collaboration is difficult with commercial closed source but we want
to keep as much of the OSS ethos as possible available to users
who want to fix it themselves.

In order to unambiguously own and sell Dradis Framework commercial
products, we must have the copyright associated with the entire
codebase. Any code you create which is merged must be owned by us.
That's not us trying to be a jerks, that's just the way it works.

Please review the [CONTRIBUTING.md](https://github.com/dradis/dradis-ce/blob/master/CONTRIBUTING.md)
file for the details.

You can delete this section, but the following sentence needs to
remain in the PR's description:

> I assign all rights, including copyright, to any future Dradis
> work by myself to Security Roots.

### CODE REVIEW

File: lib/dradis/plugins/nexpose/formats/full.rb [hunk 1]
```diff
@@ -141,15 +142,17 @@ def process_full(doc)
 
         # 3.2 associate with the nodes via Evidence.
         #   TODO: there is room for improvement here by providing proper Evidence content
-        xml_vulnerability.xpath('./hosts/host').collect(&:text).each do |host_name|
+        xml_vulnerability.xpath('./hosts/host').map(&:text).each do |host_name|
```
Comment (caitmich) [inline lib/dradis/plugins/nexpose/formats/full.rb:145]: What's the reason for this change?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,6 @@
 vX.X.X (XXX 2023)
   - Use the details in <os> as the OS node property
+  - Fix creating evidence for similar node entries
```
Comment (caitmich) [inline CHANGELOG.md:3]:
```suggestion
  - Allow multiple evidence with the same test id & node address
```
(or something more descriptive like this)

File: lib/dradis/plugins/nexpose/formats/full.rb (hunk 1 above)
Comment (aapomm) [reply lib/dradis/plugins/nexpose/formats/full.rb:145]: https://github.com/rubocop/ruby-style-guide#map-find-select-reduce-include-size although not a hard requirement
